tests/acceptance/lazy-routeable-engine-test.js

```diff
@@ -66,6 +63,40 @@ test('it should pause to load JS and CSS assets on deep link into a lazy Engine'
   andThen(() => verifyInitialBlogRoute(assert, this.loadEvents, this.application));
 });
 
+test('it should pause to load JS and CSS assets on deep link into a lazy Engine after retrying the transition', function(assert) {
+  assert.expect(7);
+
+  define('dummy/routes/application', () => (Ember.Route.extend({
+      actions: {
+        error: nope
+      }
+    })
+  ));
+
+  const adapterException = Ember.Test.adapter.exception;
+  const loader = this.application.__container__.lookup('service:asset-loader');
+  const jsLoader = loader.__assetLoaders.js;
+  const cssLoader = loader.__assetLoaders.css;
+  const failLoader = () => RSVP.reject('rejected');
+
+  Ember.Test.adapter.exception = nope;
```

Comment:
This is unfortunate, but I believe it is acceptable.  I am not sure what a valid rejection handler would actually look like in router.js (from its perspective there is no real recovery AFAICT) so doing this to avoid failing the test while we confirm the underlying behavior seems fine.

Reply:
Can you swap this to `() => {}` directly though?  Its just a little easier to grok to me...